File: backend/semgrep_rules/tests/javascript/kisa_sw_15_security_decision_input.js
function unsafe(req) {
  // ruleid: kisa.sw15.javascript.external-input-controls-security-decision
  return req.body.role === "admin";
}
function safe(req) {
  // ok: kisa.sw15.javascript.external-input-controls-security-decision
  return req.user && req.user.role === "admin";
}

File: backend/semgrep_rules/tests/python/kisa_sw_15_security_decision_input.py
def unsafe(request):
    # ruleid: kisa.sw15.python.external-input-controls-security-decision
    if request.GET.get("role") == "admin":
        return "secret"
    return "denied"

def safe(request):
    # ok: kisa.sw15.python.external-input-controls-security-decision
    if request.user.is_staff:
        return "secret"
    return "denied"
